app/models/user.rb

```diff
@@ -206,7 +207,42 @@ def self.authenticator(username = nil)
   end
 
   def self.authenticate(username, password, request = nil, options = {})
-    authenticator(username).authenticate(username, password, request, options)
+    user = authenticator(username).authenticate(username, password, request, options)
+    user.try(:link_user_to_session, request)
+
+    user
+  end
+
+  def link_user_to_session(request)
+    return unless request
+
+    session_id = request.session_options[:id]
+    sessions << Session.find_or_create_by(:session_id => session_id)
+  end
+
+  def broadcast_revoke_sessions
+    if Settings.server.session_store == "cache"
+      MiqQueue.broadcast(
+        :class_name  => self.class.name,
+        :instance_id => id,
+        :method_name => :revoke_sessions
+      )
+    else
+      # If using SQL or Memory, the sessions don't need to (or can't) be
+      # revoked via a broadcast since the session/token stores are not server
+      # specific, so execute it inline.
```

Comment:
Technically this is not true for memory, as each server will have its own dedicated session store, which means each server needs to be told to kill the sessions it knows about.  (Granted we only use memory for tests, but code-wise just in case, it can be treated the same way as memcached - I agree though that multiple UI workers will each have their own copy, and I'm not sure that's possible to handle at the moment)

Reply:
> Technically this is not true for memory

@Fryguy no, memory falls under the "can't" part because each _**process**_ owns it's own cache, so a background worker wouldn't be able to kill the cache of a server process.

I could see the argument of still allowing it though, if say we want to make sure the job got on the queue for tests.  But otherwise I don't think `.broadcast` it will function as expected with `memory` store.

Reply:
Ah yeah you're right